src/mbed_os_tools/detect/darwin.py

```diff
@@ -40,7 +42,12 @@ def _plist_from_popen(popen):
     if not out:
         return []
     try:
-        return loads(out)
+        # Beautiful soup ensures the XML is properly formed
+        soup = BeautifulSoup(out.decode('utf8'), 'xml')
```

Comment:
Is there a better name than 'soup'? It's not really clear what this code is actually doing...

Reply:
Changed the variable name